src/formatting/closures.rs

```diff
@@ -65,6 +66,33 @@ pub(crate) fn rewrite_closure(
             rewrite_closure_block(block, &prefix, context, body_shape)
         })
     } else {
+        // If there are comments between the fn decl and the body, the body (+ comments) need to be
+        // wrapped in a block. Since there's no return type annotation on closures with expr
+        // bodies, look for comments after the second "|".
+        let between_span = mk_sp(
+            context.snippet_provider.span_after(span, "|"),
+            body.span.lo(),
+        );
+        let between_span = mk_sp(
+            context.snippet_provider.span_after(between_span, "|"),
+            body.span.lo(),
+        );
```

Comment:
Instead of the multiple steps to build out this span ourselves, could we use the span associated with the param block from the closure expression? We're not currently using it but I believe it would make it easier to derive the "between span"

https://github.com/rust-lang/rustfmt/blob/f7ba2dbcf2ce9dd086c06b6b7e89b399fdd25374/src/formatting/expr.rs#L192-L196

Reply:
That turns `between_span` into:
```rust
let between_span = mk_sp(param_span.hi(), body.span.lo());
```
which is significantly cleaner, at the cost (?) of passing in one more argument. That looks much better to me!